sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2345,11 +2345,13 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       cause = null)
   }
 
-  def failedMergingSchemaError(schema: StructType, e: SparkException): Throwable = {
+  def failedMergingSchemaError(
+      leftSchema: StructType,
+      rightSchema: StructType,
+      e: SparkException): Throwable = {
     new SparkException(
-      errorClass = "_LEGACY_ERROR_TEMP_2247",
-      messageParameters = Map(
-        "schema" -> schema.treeString),
+      errorClass = "CANNOT_MERGE_SCHEMAS",
+      messageParameters = Map("left" -> leftSchema.treeString, "right" -> rightSchema.treeString),
```

Comment:
Could you wrap schemas by `toSQLType()` instead of `rightSchema`, please. BTW, the error occurs in PySpark, SQL, R, we show it in some common form as a SQL type.

Reply:
Thanks for the explanation. I have wrapped the schemas with `toSQLType`